### DISCUSSION TITLE
Vector Internal log level change

### DISCUSSION THREAD

**Discussion by bmallya** [vectordotdev/vector#18418] [General]
Hello,
Is there a way to change the log level of a running instance of vector? I might want to start off with logging only WARN and above, but eventually want to change it to verbose.

Thanks

**Comment by jszwedko** [comment]
Hi @bmallya !

Not directly, though I think there is a workaround you could use (see below). Would you want to open a feature request? That'll help us track demand for this feature.

The workaround I'm thinking of is, rather than consuming the logs from stdout/stderr, route those to `/dev/null`, and then use the `internal_logs` source to consume Vector's logs at the `INFO` level but then feed it through a `filter` transform to filter out anything less severe than `WARN`. The `filter` component could be dynamically reloaded to change the log level by updating the configuration file and issuing a SIGHUP to Vector.
